pkg/osquery/extension.go

```diff
@@ -426,38 +425,28 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 		return "", true, fmt.Errorf("generating UUID: %w", err)
 	}
 
-	// We used to see the enrollment details fail, but now that we're running as an exec,
-	// it seems less likely. Try a couple times, but backoff fast.
-	enrollDetails := getRuntimeEnrollDetails()
-	if osqPath := e.knapsack.LatestOsquerydPath(ctx); osqPath == "" {
-		e.slogger.Log(ctx, slog.LevelInfo,
-			"skipping enrollment osquery details, no osqueryd path, this is probably CI",
-		)
-		span.AddEvent("skipping_enrollment_details")
-	} else {
-		if err := backoff.WaitFor(func() error {
-			err = getOsqEnrollDetails(ctx, osqPath, &enrollDetails)
-			if err != nil {
-				e.slogger.Log(ctx, slog.LevelDebug,
-					"getOsqEnrollDetails failed in backoff",
-					"err", err,
-				)
-			}
-			return err
-		}, 30*time.Second, 5*time.Second); err != nil {
-			if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_REQUIRED") == "true" {
-				return "", true, fmt.Errorf("query osq enrollment details: %w", err)
-			}
+	var enrollDetails types.EnrollmentDetails
 
-			e.slogger.Log(ctx, slog.LevelError,
-				"failed to get osq enrollment details with retries, moving on",
-				"err", err,
-			)
-			traces.SetError(span, fmt.Errorf("query osq enrollment details: %w", err))
-		} else {
-			span.AddEvent("got_enrollment_details")
+	err = backoff.WaitFor(func() error {
+		details := e.knapsack.GetEnrollmentDetails()
+		if err != nil {
+			traces.SetError(span, fmt.Errorf("error getting enrollment details: %w", err))
+			return err
+		}
```

Comment:
I think this is checking the err from `e.getHostIdentifier()` on 423? While it doesn't make any practical difference to the code, I think it's easier to read if we remove the double errcheck.

```suggestion
```